b2share/modules/records/triggers.py

```diff
@@ -94,6 +90,11 @@ def set_record_owner(record, **kwargs):
     record['owner'] = str(current_user.id)
 
 
+def add_oai_id(record):
```

Comment:
What about using PID minter (https://github.com/inveniosoftware/invenio-oaiserver/blob/master/invenio_oaiserver/minters.py) or defining your own? Each OAIID should be registered.